applications/ConvectionDiffusionApplication/custom_strategies/strategies/explicit_runge_kutta_4_eulerian_convdiff_strategy.h

```diff
@@ -0,0 +1,385 @@
+    {
+        rOStream << Info();
+    }
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    /**
+     * @brief Initialize the Runge-Kutta substep.
+     * @details Calculate the orthogonal subscale projections if required.
+     */
+    virtual void InitializeRungeKuttaIntermediateSubStep() override
+    {
+        KRATOS_TRY;
+
+        BaseType::InitializeRungeKuttaIntermediateSubStep();
+
+        auto& r_model_part = BaseType::GetModelPart();
+        auto& r_process_info = r_model_part.GetProcessInfo();
+        // set interpolation parameter TIME_INTEGRATION_THETA
+        if (r_process_info.GetValue(RUNGE_KUTTA_STEP) == 1) {
+            r_process_info.GetValue(TIME_INTEGRATION_THETA) = 1.0;
+        }
+        else if (r_process_info.GetValue(RUNGE_KUTTA_STEP) == 2 || r_process_info.GetValue(RUNGE_KUTTA_STEP) == 3) {
+            r_process_info.GetValue(TIME_INTEGRATION_THETA) = 0.5;
+        }
+        else {
+            KRATOS_ERROR << "RUNGE-KUTTA step not correctly set.";
+        }
+        // execute OSS step, if needed
+        if (r_process_info[OSS_SWITCH] == 1) {
+            ExecuteOSSStep();
+        }
+
+        KRATOS_CATCH("");
+    };
+
+    /**
+     * @brief Initialize the Runge-Kutta substep.
+     * @details Calculate the orthogonal subscale projections if required.
+     */
+    virtual void InitializeRungeKuttaLastSubStep() override
+    {
+        KRATOS_TRY;
+
+        BaseType::InitializeRungeKuttaLastSubStep();
+        auto& r_model_part = BaseType::GetModelPart();
+        auto& r_process_info = r_model_part.GetProcessInfo();
+        // set interpolation parameter TIME_INTEGRATION_THETA
+        if (r_process_info.GetValue(RUNGE_KUTTA_STEP) == 4) {
+            r_process_info.GetValue(TIME_INTEGRATION_THETA) = 0.0;
+        }
+        else {
+            KRATOS_ERROR << "RUNGE-KUTTA step not correctly set.";
+        }
+        // execute OSS step, if needed
+        if (r_process_info[OSS_SWITCH] == 1) {
+            ExecuteOSSStep();
+        }
+
+        KRATOS_CATCH("");
+    };
+
+    /**
+     * @brief Finalize the Runge Kutta explicit solver step.
+     * @details Calculate the orthogonal subscale projections if required.
+     */
+    virtual void FinalizeSolutionStep() override
+    {
+        KRATOS_TRY;
+
+        auto& r_model_part = BaseType::GetModelPart();
+        auto& r_process_info = r_model_part.GetProcessInfo();
+        // set interpolation parameter TIME_INTEGRATION_THETA
+        r_process_info.GetValue(TIME_INTEGRATION_THETA) = 0.0;
+        // execute OSS step, if needed
+        if (r_process_info[OSS_SWITCH] == 1) {
+            ExecuteOSSStep();
+        }
+        BaseType::FinalizeSolutionStep();
```

Comment:
The base calls must be executed prior to any operation.

Reply:
I first need to compute OSS subscales, and then call the FinalizeSolutionStep() of the element (which updates the dynamic subscales value according to OSS values).

Reply:
Then shouldn't the OSS be computed at the end of the `SolveSolutionStep`?

Reply:
Well ok, but we will need to check that we are both in the last RK step and we are using OSS. But ok, I will make the change.
Can I ask why it is important to have the call to the base method before to any other operation?

Reply:
Done the change.

Reply:
Because your customized function might depend on something that has been done by the base class implementation (but not in the other way around).